src/control/lib/control/pool.go

```diff
@@ -776,94 +777,73 @@ func (resp *PoolRanksResp) Errors() error {
 	if resp == nil {
 		return errors.Errorf("nil %T", resp)
 	}
-	if resp.Status == int32(daos.Success) {
-		return nil
+	if resp.ID == "" {
+		return errors.New("empty id in response")
```

Comment:
Similar comment about erroring on the response returned to us from the engine. If the engine returns something bad, I think we should still make an attempt to return/display as much info as possible to the user. I.e. if the ID is unknown, we should still try to return the rank count errors for an "unknown pool".